tests/robustness/snapmeta/simple.go

```diff
@@ -28,19 +29,43 @@ func NewSimple() *Simple {
 	}
 }
 
-// Store implements the Storer interface Store method.
-func (s *Simple) Store(key string, val []byte) error {
-	buf := make([]byte, len(val))
-	_ = copy(buf, val)
+// IndexOperation defines several operations implemented on Index.
+type IndexOperation string
 
+const (
+	// AddToIndexOperation add value in Index.
+	AddToIndexOperation IndexOperation = "addToIndex"
+	// RemoveFromIndexOperation removes value from Index.
+	RemoveFromIndexOperation IndexOperation = "removeFromIndex"
+)
+
+// Store implements the Storer interface Store method.
+func (s *Simple) Store(key string, val []byte, indexUpdates map[string]IndexOperation) error {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
+	buf := make([]byte, len(val))
+
+	_ = copy(buf, val)
```

Comment:
The copy doesn't need to be inside the lock, does it? We can keep the lock open while doing this local work